components/script/dom/bindings/reflector.rs

```diff
@@ -23,7 +23,16 @@ where
     U: DerivedFrom<GlobalScope>,
 {
     let global_scope = global.upcast();
-    unsafe { T::WRAP(GlobalScope::get_cx(), global_scope, obj) }
+    unsafe { T::WRAP(GlobalScope::get_cx(), global_scope, None, obj) }
+}
+
+pub fn reflect_dom_object2<T, U>(obj: Box<T>, global: &U, proto: Option<HandleObject>) -> DomRoot<T>
```

Comment:
Nit: Would it be possible to name this `reflect_dom_object_with_proto` so that it follows a similar pattern with `reflect_node_with_proto`?

Another thing to consider is to remove the `with_proto` variants entirely and always pass the prototype, but it seems like the large amount of noisy changes needed for that would be better suited for a possible followup.